src/dp_vnf.c

```diff
@@ -89,83 +156,52 @@ int dp_del_vnf(const uint8_t ul_addr6[DP_VNF_IPV6_ADDR_SIZE])
 		return ret;
 	}
 
-	rte_free(vnf);
-
 	ret = rte_hash_del_key(vnf_handle_tbl, ul_addr6);
 	if (DP_FAILED(ret)) {
 		DPGRPC_LOG_WARNING("Cannot remove VNF", DP_LOG_IPV6(ul_addr6), DP_LOG_RET(ret));
 		return ret;
 	}
 
+	ret = dp_delete_vnf_value(vnf);
+	if (DP_FAILED(ret)) {
+		DPGRPC_LOG_WARNING("Cannot delete VNF value record");
+		return ret;
+	}
+
+	rte_free(vnf);
```

Comment:
I think you also need to free this when `dp_delete_vnf_value()` fails, because the key is already gone from the `vnf_handle_tbl` thus making this an orphan pointer.

Reply:
done.

Reply:
You can make it neater if you free right before the DP_FAILED condition.